runtime/Java/src/org/antlr/v4/runtime/atn/ATNDeserializer.java

```diff
@@ -90,6 +90,12 @@ public ATN deserialize(char[] data) {
 
 		int p = 0;
 		int version = toInt(data[p++]);
+    if (version == LEGACY_SERIALIZED_VERSION) {
+      // Preserve backwards compatibility for version 3. We simply skip over the UUID and assume all
+      // features were supported.
+      p += 8;
+      version = SERIALIZED_VERSION;
+    }
```

Comment:
Use tabs instead of spaces (ANTLR uses tabs formatting).

Reply:
Yeah,I am abandoning backward compatibility except for the java target because we have special internal problem to resolve. I'll deal with this when I can look at the 32 bit implementation.

Reply:
Yeah, I guess I will do a bit of copy and paste programming for java if I get the 32-bit deserialization going.  thanks for pointing this out